lib/ex_webrtc/recorder/converter.ex

```diff
@@ -37,105 +46,324 @@ defmodule ExWebRTC.Recorder.Converter do
     channels: 2
   }
 
-  @default_output_path "./converter_output"
+  @default_output_path "./converter/output"
+  @default_download_path "./converter/download"
+  @default_thumbnail_width 640
+  @default_thumbnail_height -1
+
+  @typedoc """
+  Context for the thumbnail generation.
+
+  * `:width` - Thumbnail width. #{@default_thumbnail_width} by default.
+  * `:height` - Thumbnail height. #{@default_thumbnail_height} by default.
+
+  Setting either of the values to `-1` will fit the size to the aspect ratio.
+  """
+  @type thumbnails_ctx :: %{
+          optional(:width) => pos_integer() | -1,
+          optional(:height) => pos_integer() | -1
+        }
+
+  @typedoc """
+  Options that can be passed to `convert_manifest!/2` and `convert_path!/2`.
+
+  * `:output_path` - Directory where Converter will save its artifacts. `#{@default_output_path}` by default.
+  * `:s3_upload_config` - If passed, processed recordings will be uploaded to S3-compatible storage.
+    See `t:ExWebRTC.Recorder.S3.upload_config/0` for more info.
+  * `:download_path` - Directory where Converter will save files fetched from S3. `#{@default_download_path}` by default.
+  * `:s3_download_config` - Optional S3 config overrides used when fetching files.
+    See `t:ExWebRTC.Recorder.S3.override_config/0` for more info.
+  * `:thumbnails_ctx` - If passed, Converter will generate thumbnails for the output files.
+    See `t:thumbnails_ctx/0` for more info.
+  """
+  @type option ::
+          {:output_path, Path.t()}
+          | {:s3_upload_config, keyword()}
+          | {:download_path, Path.t()}
+          | {:s3_download_config, keyword()}
+          | {:thumbnails_ctx, thumbnails_ctx()}
+
+  @type options :: [option()]
 
   @doc """
-  Convert the saved dumps of tracks in the report to IVF and Ogg files.
+  Loads the recording manifest from file, then proceeds with `convert_manifest!/2`.
   """
-  @spec convert!(Path.t(), Path.t()) :: :ok | no_return()
-  def convert!(report_path, output_path \\ @default_output_path) do
-    report_path =
-      report_path
+  @spec convert_path!(Path.t(), options()) :: __MODULE__.Manifest.t() | no_return()
+  def convert_path!(recorder_manifest_path, options \\ []) do
```

Comment:
Nitpick: I would call this convert and use pattern matching to see whether we got path or manifest as an argument